src/test/tree/EigenPodManager.tree

```diff
@@ -0,0 +1,95 @@
+├── when setMaxPods is called
+│   ├── given the user is not the pauser
+│   │   └── it should revert
+│   └── given the user is the pauser
+│       └── it should set the max pods
+├── when updateBeaconChainOracle is called
+│   ├── given the user is not the owner
+│   │   └── it should revert
+│   └── given the user is the owner
+│       └── it should set the beacon chain oracle
+├── when addShares is called
+│   ├── given that the caller is not the delegationManager
+│   │   └── it should revert
+│   ├── given that the podOwne address is 0
+│   │   └── it should revert
+│   ├── given that the shares amount is negative
+│   │   └── it should revert
+│   ├── given that the shares is not a whole gwei amount
+│   │   └── it should revert
+│   ├── given that too many shares are withdrawn
+│   │   └── it should revert
+│   └── given that all of the above conditions are satisfied
+│       └── it should update the podOwnerShares
+├── when removeShares is called
+│   ├── given that the caller is not the delegationManager
+│   │   └── it should revert
+│   ├── given that the shares amount is negative
+│   │   └── it should revert
+│   ├── given that the shares is not a whole gwei amount
+│   │   └── it should revert
+│   ├── given that too many shares are removed
+│   │   └── it should revert
+│   └── given that all of the above conditions are satisfied
+│       └── it should update the podOwnerShares
+├── when shares are withdrawn as tokens
+│   ├── given that the podOwner is address 0
+│   │   └── it should revert
+│   ├── given that the destination is address 0
+│   │   └── it should revert
+│   ├── given that the shares amount is negative
+│   │   └── it should revert
+│   ├── given that the shares is not a whole gwei amount
+│   │   └── it should revert
+│   ├── given that the current podOwner shares are negative
+│   │   ├── given that the shares to withdraw are greater than the negative shares of the podOwner
+│   │   │   └── it should set the podOwnerShares to 0 and decrement shares to withdraw by the share defecit
+│   │   └── given that the shares to withdraw are less than the negative shares of the podOwner
+│   │       └── it should increment the podOwner shares by the shares to withdraw
+│   └── given that the pod owner shares are positive
+│       └── it should withdraw restaked ETH from the eigenPod
+├── when shares are adjusted ***
+│   ├── given that sharesBefore is negative or 0
+│   │   ├── given that sharesAfter is negative or zero
+│   │   │   └── the change in delegateable shares should be 0
+│   │   └── given that sharesAfter is positive
+│   │       └── the change in delegateable shares should be positive
+│   └── given that sharesBefore is positive
+│       ├── given that sharesAfter is negative or zero
+│       │   └── the change in delegateable shares is negative sharesBefore
+│       └── given that sharesAfter is positive
+│           └── the change in delegateable shares is the difference between sharesAfter and sharesBefore
```

Comment:
feels like a description of the paths in the internal `_calculateChangeInDelegatableShares` function -- should we perhaps just use wrappers that expose these internal functions for unit testing (e.g. `contract EigenPodManagerWrapper is EigenPodManager ...`), and include these paths in the tree structure diagrams in general?

Reply:
Good suggestion, was planning on testing in integration but this is a better way to do it